app/views/layouts/_asset_reconciliation.html.erb

```diff
@@ -0,0 +1,65 @@
+
+    <meta name="proper-stylesheet-crayons" content="<%= stylesheet_url "crayons" %>">
+    <meta name="proper-stylesheet-views" content="<%= stylesheet_url "views" %>">
+    <meta name="proper-stylesheet-minimal" content="<%= stylesheet_url "minimal" %>">
+    <meta name="proper-js-base" content="<%= javascript_url "base" %>">
+
+    <script async>
+
+      // This area is designed to reconcile issues of mismatched caches stylesheets.
+      // Currently it handles a scenario where the head assets are out of date and need to be updated.
+      // There is a legitimate reverse scenario we do not yet handle where we *should* use the "out-of-date"
+      // content on the page, but since we may have issues persisting old stylesheets etc.
+      // this only handles the most troubling issues of pages that load with no styles.
+
+      var headCacheCheck = document.querySelector("meta[name='head-cached-at']");
+      var pageCacheCheck = document.querySelector("meta[name='page-cached-at']");
+      var headCrayonsPath = document.getElementById("main-crayons-stylesheet")
+      var headViewsPath = document.getElementById("main-views-stylesheet")
+      var headMinimalPath = document.getElementById("main-minimal-stylesheet")
+      var pageCrayonsPath = document.querySelector("meta[name='proper-stylesheet-crayons']");
+      var pageViewsPath = document.querySelector("meta[name='proper-stylesheet-views']");
+      var pageMinimalPath = document.querySelector("meta[name='proper-stylesheet-minimal']");
+
+      var headBaseJSPath = document.getElementById("main-base-javascript");
+      var pageBaseJSPath = document.querySelector("meta[name='proper-js-base']");
+
+      if (headCacheCheck && headCrayonsPath && // If these elements exist, we can proceed.
+        parseInt(pageCacheCheck.content) > parseInt(headCacheCheck.content) && // If page cached-at is larger than head cached-at
+        (pageCrayonsPath.content != headCrayonsPath.href || // If the head-cached values do not match page-cached, we replace
+         pageViewsPath.content != headViewsPath.href ||
+         pageMinimalPath.content != headMinimalPath.href ||
+         pageBaseJSPath.content != headBaseJSPath.href)
+      ) {
+
+        var head = document.head;
+
+        // Add new stylesheets if we deem we should.
+        var link = document.createElement("link");
+        link.type = "text/css";
+        link.rel = "stylesheet";
+        link.href = pageCrayonsPath.content;
+        head.appendChild(link);
+        var link = document.createElement("link");
+        link.type = "text/css";
+        link.rel = "stylesheet";
+        link.href = pageViewsPath.content;
+        head.appendChild(link);
+        var link = document.createElement("link");
+        link.type = "text/css";
+        link.rel = "stylesheet";
+        link.href = pageMinimalPath.content;
+        head.appendChild(link);
+
+        // Take a breath and then remove the old ones.
```

Comment:
Thanks for adding these comments!! I think they're super useful. What's the reason for "taking the breath" with `setTimeout` here? It's not immediately obvious to me, so it might be nice to include that in the comment since you've taken the time to document it all!